### ISSUE TITLE
[Bug]: @storybook/react-vite has a dependency conflict with Vite v5

### ISSUE THREAD

**Issue by abogaart** [storybookjs/storybook#25125]
### Describe the bug

When installing Storybook in a React-Vite5 environment, the following warning is shown:
```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @vitejs/plugin-react@3.1.0
npm WARN Found: vite@5.0.5
npm WARN node_modules/vite
npm WARN   dev vite@"^5.0.0" from the root project
npm WARN   4 more (@joshwooding/vite-plugin-react-docgen-typescript, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer vite@"^4.1.0-beta.0" from @vitejs/plugin-react@3.1.0
npm WARN node_modules/@storybook/react-vite/node_modules/@vitejs/plugin-react
npm WARN   @vitejs/plugin-react@"^3.0.1" from @storybook/react-vite@7.6.3
npm WARN   node_modules/@storybook/react-vite
npm WARN
npm WARN Conflicting peer dependency: vite@4.5.1
npm WARN node_modules/vite
npm WARN   peer vite@"^4.1.0-beta.0" from @vitejs/plugin-react@3.1.0
npm WARN   node_modules/@storybook/react-vite/node_modules/@vitejs/plugin-react
npm WARN     @vitejs/plugin-react@"^3.0.1" from @storybook/react-vite@7.6.3
npm WARN     node_modules/@storybook/react-vite
```

The `@storybook/react-vite` package has a `peerDependency` on Vite that allows for versions 3, 4 and 5, but it is pinned to version ^4.1.0-beta.0 as that is a `peerDependency` of dependency `@vitejs/plugin-react`.

The simplest solution would be to bump `@vitejs/plugin-react` to the latest (4.2.1), but the release notes state the following since v4:

- Vite 4.2 is required as a peer dependency (https://github.com/vitejs/vite-plugin-react/pull/128)

So that might be an issue regarding the `peerDependencies` of `@storybook/react-vite` and Vite v3.

### To Reproduce

```
npm init vite@latest my-react-demo-project -- --template react-ts
cd my-react-demo-project
npm install @storybook/react-vite --save-dev
npm ci
```

**Comment by mjgerace** [comment]
Also have this issue, should be a straightforward peer dep clarification

**Comment by thenick775** [comment]
Bump, also would love to get this fixed!

**Comment by TomDUVAL-MAHE** [comment]
Hello !
I have this error too, do you have any idea ?

**Comment by valentinpalkovic** [comment]
We are actively working on this. In Storybook 8.0 we will completely remove `@vitejs/plugin-react` as a direct dependency.

**Comment by MichalKurzewski** [comment]
can confirm. downgrading vite to 4.5.1 seems to "fix" it.

**Comment by kasperpeulen** [comment]
Will be fixed in this PR:
https://github.com/storybookjs/storybook/pull/25427

**Comment by vanessayuenn** [comment]
This is fixed and released in the latest 8.0-alpha
